### DISCUSSION TITLE
SIP-3 (Transaction Insights) Discussion

### DISCUSSION THREAD

**Discussion by hmalik88** [MetaMask/SIPs#31] [SIPs]
This thread is meant to be a general place for open discussion concerning the state of [SIP-3 (Transaction Insights)](https://github.com/MetaMask/SIPs/blob/main/SIPS/sip-3.md).

The SIP is meant to define the transaction insight API

**Comment by ritave** [comment]
How should the insights interact with the UI in terms of blocking?

Say an insight takes 30 seconds to calculate, will that mean that the pop-up won't show up during that time, or will the insight tab have a loading indicator?

**Reply by hmalik88** [reply]
The insight tab will have a loading indicator

**Comment by ritave** [comment]
I think extending insights into message signing would be useful as well.

For example, we could have an insight that checks `eth_signTypedData` message for inclusion of a nonce property and warn that this kind of signature can result in replay attacks.

**Comment by christian-forta** [comment]
This is a great proposal. Can you elaborate on whether the tx information that is passed along also contains information about the URL of the Dapp? I think this would be extremely useful in context of a security snap plug in that warns users from txs that relate to scams/ phishing attacks.

**Reply by Montoya** [reply]
Hi Christian, the tx object does not contain information about the referring dapp. The tx-insights API is not meant to be used to get that information anyway. There may be an API in the future that would allow this, but we do not have plans to implement it yet.
